crates/bvs-vault-router/src/contract.rs

```diff
@@ -181,6 +191,140 @@ mod execute {
+                return Err(InvalidSlashingRequest {
+                    msg: "Service has not enabled slashing at timestamp.".to_string(),
+                })
+            }
+        };
+
+        // ensure bips must not exceed max_slashing_bips set by service
+        if data.bips > slashing_parameters.max_slashing_bips {
+            return Err(InvalidSlashingRequest {
+                msg: "Slashing bips is over max_slashing_bips set.".to_string(),
+            });
+        }
+
+        // ensure that the operator has opted in to slashing at the given timestamp.
+        let IsOperatorOptedInToSlashingResponse(is_operator_opted_in) =
+            deps.querier.query_wasm_smart(
+                registry,
+                &bvs_registry::msg::QueryMsg::IsOperatorOptedInToSlashing {
+                    service: service.to_string(),
+                    operator: operator.to_string(),
+                    timestamp: Some(data.timestamp.seconds()),
+                },
+            )?;
+
+        if !is_operator_opted_in {
+            return Err(InvalidSlashingRequest {
+                msg: "Operator has not opted-in to slashing at timestamp.".to_string(),
+            });
+        }
+
+        // get current active slashing request for (service, operator) pair
+        let active_slashing_requests =
+            state::get_active_slashing_requests(deps.storage, &service, &operator)?;
+
+        // if active slashing request exists and
+        // if request_expiry > now (in the future) => request hasn't expired,
+        // so the service has to manually cancel active request (throw Err)
+        if let Some(active_slashing_requests) = active_slashing_requests {
+            if active_slashing_requests.request_expiry > env.block.time {
+                return Err(InvalidSlashingRequest {
+                    msg: "Service has current active slashing request for the operator."
+                        .to_string(),
+                });
+            }
+        }
+
+        // else, it will be overriden by the current slash request
+
+        // New_request_expiry will be using `resolution_window`
+        // value from the timestamp's slashing_parameters,
+        // instead of the most recent slashing param.
+        // This ensures that both parties agree upon all parameters used.
+        let new_request_expiry = env
+            .block
+            .time
+            .plus_seconds(slashing_parameters.resolution_window * 2);
+        let new_request = SlashingRequest {
+            request: data.clone(),
+            request_time: env.block.time,
+            request_expiry: new_request_expiry,
+        };
+
+        // save slash data
+        let slashing_request_id =
+            state::save_slashing_request(deps.storage, &service, &operator, &new_request)?;
+
+        Ok(Response::new()
+            .set_data(RequestSlashingResponse(slashing_request_id.clone()))
+            .add_event(
+                Event::new("SlashingRequest")
```

Comment:
```suggestion
                Event::new("RequestSlashing")
```

Should we follow the enum?